### PR TITLE
Adds User State using context api

### PR DESCRIPTION
**Pull request by sourabhmodi20051**

### CODE REVIEW

File: client/src/components/AuthForm/AuthForm.js [hunk 1]
```diff
@@ -51,27 +56,26 @@ const AuthForm = (props) => {
           />
         </figure>
 
-        {
-          inputs.map(({type}) => renderInputField({type, key: type}))
-        }
+        {inputs.map(({ type }) => renderInputField({ type, key: type }))}
```
Comment (nimish-gupta) [inline client/src/components/AuthForm/AuthForm.js]: Key is type, pretty weird, you can add the index..

File: client/src/index.js [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
-      <AuthRoute exact path="/signup" component={SignUp} />
-      <PrivateRoute exact path="/workspaces" component={Workspaces} />
-      <PrivateRoute exact path="/thread" component={Thread} />
-      <Route component={NotFound} />
-    </Switch>
-  </Router>
+  <AppContextProvider>
+    <Router>
+      <Switch>
+        {/* <PrivateRoute exact path='/' Redirect to="/workspaces" />)} /> */}
```
Comment (nimish-gupta) [inline client/src/index.js]: Remove comments.

File: client/src/containers/SignIn/utils.js [hunk 3]
```diff
@@ -1,17 +1,18 @@
 import callApi from '../../libs/axios';
 
-export const handleSubmit = async (e, user, history) => {
+export const handleSubmit = async (e, user, history, dispatch) => {
   e.preventDefault();
   const { email, password } = user;
   const result = await callApi('post', '/user/login', { email, password });
   if (result.data) {
     const { Data } = result.data;
```
Comment (nimish-gupta) [inline client/src/containers/SignIn/utils.js]: Data in capital letters? variable names should be in small camelCase form while Components should be in upperCase Form.

File: client/src/components/AuthForm/AuthForm.js (hunk 1 above)
Comment (sourabhmodi20051) [reply client/src/components/AuthForm/AuthForm.js]: Added index here

File: client/src/containers/SignIn/utils.js (hunk 3 above)
Comment (sourabhmodi20051) [reply client/src/containers/SignIn/utils.js]: changed to `data`

File: client/src/index.js (hunk 2 above)
Comment (sourabhmodi20051) [reply client/src/index.js]: Removed

**Comment by sourabhmodi20051** [comment]
> Please add gif/video with the changes.

This PR doesn't add any new feature. It refactors the code to get user state using context api

**Pull request merged**